src/app/security/maturity-model/components/TableOfContents.tsx

```diff
@@ -0,0 +1,55 @@
+'use client'
+
+import { clsx } from 'clsx'
+import { useIntersectionObserver } from 'usehooks-ts'
+
+import { coreFunctionsData } from '../data/coreFunctionsData'
+import { scrollToSection } from '../data/scrollToSection'
+import {
+  useSectionHashObserver,
+  HASH_SIGN,
+  type SectionHash,
+} from '../utils/useSectionHashObserver'
+
+export function TableOfContents() {
+  const { currentHash, clearSectionHash } = useSectionHashObserver()
+
+  const { ref } = useIntersectionObserver({
+    onChange: (isIntersecting) => {
+      if (!isIntersecting) {
+        clearSectionHash()
+      }
+    },
+  })
+
+  return (
+    <nav ref={ref}>
+      <p className="mb-4 ml-6 text-sm font-semibold uppercase text-brand-200 lg:mb-6">
+        Table of Contents
+      </p>
+
+      <ul>
+        {coreFunctionsData.map(({ slug, title }, index) => {
```

Comment:
We should maybe reconsider markup https://css-tricks.com/a-perfect-table-of-contents-with-html-css/

Reply:
Good idea! @CharlyMartin